protocol-correctness/lib/functions/map-values-become-keys.k

```diff
@@ -0,0 +1,30 @@
+require "protocol-correctness/lib/functions/keys-map.k"
+
+module MAP-VALUES-BECOME-KEYS-SYNTAX
+  imports BOOL
+  imports MAP
+
+  syntax Bool ::= mapValuesBecomeKeys(Map, Map)  [function, functional]
+endmodule
+
+module MAP-VALUES-BECOME-KEYS-ADDITIONAL-SYNTAX
+  imports BOOL
+  imports MAP
+
+  syntax Bool ::= #mapValuesBecomeKeys(Map, Map)  [function, functional]
+endmodule
+
+module MAP-VALUES-BECOME-KEYS
+  imports MAP-VALUES-BECOME-KEYS-ADDITIONAL-SYNTAX
+  imports MAP-VALUES-BECOME-KEYS-SYNTAX
+
+  imports KEYS-MAP-SYNTAX
+
+  rule mapValuesBecomeKeys(M:Map, N:Map) => #mapValuesBecomeKeys(M, keysMap(N))
```

Comment:
Why do you need to turn map N into keysMap, which makes all the values equal to 0?

Reply:
Ok. Can you add a comment in the file?

Reply:
Done.